numpy/lib/stride_tricks.py

```diff
@@ -111,6 +112,147 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
     return view
 
 
+def _sliding_window_view_dispatcher(x, shape, axis=None, *,
+                                    subok=None, writeable=None):
+    return (x,)
+
+
+@array_function_dispatch(_sliding_window_view_dispatcher, module='numpy')
+def sliding_window_view(x, shape, axis=None, *, subok=False, writeable=False):
+    """
+    Create a sliding window view into the array with the given shape.
+
+    Creates a sliding window view of the N dimensional array with the given
+    window shape. Window slides across each dimension of the array and extract
+    a subsets of the array at any window position.
+
+    Parameters
+    ----------
+    x : array_like
+        Array to create the sliding window view from.
+
+    shape : int or tuple of int
+        The shape of the window. If `axis` is not present, must have same
+        length as the number of input array dimensions.
+        Single integers `i` are treated as if they were the tuple `(i,)`.
+
+    axis : int or tuple of int, optional
+        If present, `shape[i]` will refer to the axis `axis[i]` of `x`,
+        otherwise `shape[i]` will refer to axis `i` of `x`.
+        Single integers `i` are treated as if they were the tuple `(i,)`.
+
+    subok : bool, optional
+        If True, sub-classes will be passed-through, otherwise the returned
+        array will be forced to be a base-class array (default).
+
+    writeable : bool, optional
+        When true, allow writing to the returned view. The default is false,
+        as this should be used with caution: the returned view contains the
+        same memory location multiple times, so writing to one location will
+        cause others to change.
+
+    Returns
+    -------
+    view : ndarray
+        Sliding window view of the array.
+
+    See Also
+    --------
+    as_strided: Create a view into the array with the given shape and strides.
+    broadcast_to: broadcast an array to a given shape.
+
+    Notes
+    -----
+    For some cases there may be more efficient approaches to calculate
+    transformations across multi-dimensional arrays, for instance
+    `scipy.signal.fftconvolve`, where combining the iterating step with the
+    calculation itself while storing partial results can result in significant
+    speedups.
+
+    Examples
+    --------
+    >>> x = np.arange(5)
+    >>> sliding_window_view(x, 2)
+    array([[0, 1],
+           [1, 2],
+           [2, 3],
+           [3, 4]])
```

Comment:
In my opinion this is slightly easier to understand with a bigger array,
```suggestion
    >>> x = np.arange(6)
    >>> sliding_window_view(x, 3)
    array([[0, 1, 2],
           [1, 2, 3],
           [2, 3, 4],
           [3, 4, 5]])
```